### DISCUSSION TITLE
Welcome to MaRGE Discussions!

### DISCUSSION THREAD

**Discussion by josalggui** [josalggui/MaRGE#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by xiangzui** [comment]
Hello, after I clicked on Autocalibration and Localizer, the Acquire Button turned grey and became unusable.

**Reply by xiangzui** [reply]
what should i do ?

**Reply by volatile-static** [reply]
Just wait for the acquisition.

**Reply by xiangzui** [reply]
I waited for more than ten minutes, but it still didn't work

**Reply by volatile-static** [reply]
you should debug the python code to see what's wrong with the acquisition procedure

**Reply by josalggui** [reply]
did you configure the localizer? the localizer should be a fast sequence.

**Reply by xiangzui** [reply]
yes i did. Is this error related to encoding? In Python, the error 'gbk' codec can't encode character '\ xba' in position 73: illegal multibyte sequence

**Reply by josalggui** [reply]
not sure, I did not get that error before. One general suggestion. Open the GUI, click on the M button to do the copy_bitstream, then connect to the server. Once you are connected to the server, in the custom tab select noise and run it. Let me know the result

**Reply by volatile-static** [reply]
To fix GBK-related errors: https://github.com/volatile-static/MaRGEx/commit/480c78c37cd51917f66b01a41c63a3be38fdd647

**Reply by xiangzui** [reply]
<img width="1280" alt="noise" src="https://github.com/user-attachments/assets/0463131e-4dab-4153-9068-2587614c783f">
Okay, this is the result.Nosie and FID can successfully represent it, but if GRE are selected, the signal cannot be displayed

**Reply by xiangzui** [reply]
> not sure, I did not get that error before. One general suggestion. Open the GUI, click on the M button to do the copy_bitstream, then connect to the server. Once you are connected to the server, in the custom tab select noise and run it. Let me know the result↳
Okay, this is the result. Nosie and FID can successfully represent it, but if  GRE are selected, the signal cannot be displayed.

**Reply by xiangzui** [reply]
> To fix GBK-related errors: [volatile-static@480c78c](https://github.com/volatile-static/MaRGEx/commit/480c78c37cd51917f66b01a41c63a3be38fdd647)↳

thank u!

**Comment by xiangzui** [comment]
When running the GRE, the following message will appear: Sequence waveforms loaded successfully
{'warnings': ['ocra1 data was lost (overwritten before being sent)']}
<img width="1280" alt="gre" src="https://github.com/user-attachments/assets/2a076040-7d51-4ddb-951f-c4c6f1583d19">

**Reply by josalggui** [reply]
The warning you see is normal. If I remember, it was related to repeated data in the waveforms. The reason gre1d does not work is that it was outdated. I fixed the bug and tested in my computer. Now looks working. Just update to the last commit.

**Reply by xiangzui** [reply]
> The warning you see is normal. If I remember, it was related to repeated data in the waveforms. The reason gre1d does not work is that it was outdated. I fixed the bug and tested in my computer. Now looks working. Just update to the last commit.

ok,thank u.
In addition, can this software simulate the timing diagram of the signal?

**Reply by josalggui** [reply]
You can see the time diagram by clicking in the corresponding button of the toolbar. https://github.com/josalggui/MaRGE/wiki/Toolbars

**Reply by xiangzui** [reply]
> You can see the time diagram by clicking in the corresponding button of the toolbar. https://github.com/josalggui/MaRGE/wiki/Toolbars

okok，thank u.Now,i already solve "the Acquire Button turned grey and became unusable",The reason is due to issues with my IDE settings. Additionally, may I ask if it is possible to change the type of radio frequency (RF) pulse?

**Reply by josalggui** [reply]
The available sequences currently only use square pulses as there is no slice selection. However, mriBlankSeq includes a method called "rfSincPulse" to create sinc pulses. Additionally, you can create your own methods in mriBlankSeq class to run custom RF pulse shapes with modulated amplitude or phase.

**Reply by xiangzui** [reply]
> The available sequences currently only use square pulses as there is no slice selection. However, mriBlankSeq includes a method called "rfSincPulse" to create sinc pulses. Additionally, you can create your own methods in mriBlankSeq class to run custom RF pulse shapes with modulated amplitude or phase.↳

I found 'sequencesCalibration. py' and found the SliceSelection function in this file, but I haven't figured out how to use it yet. Has this feature not been applied yet? Also, is there no function to adjust the number of phase codes and frequency codes yet?

**Reply by josalggui** [reply]
sequenceCalibration.py is an outdated file from the early stages of the GUI and is no longer in use. I will delete it at some point. SliceSelection was an initial attempt at implementing slice selection, but I did not continue with that approach. Apologies for any confusion.

**Reply by xiangzui** [reply]
> sequenceCalibration.py is an outdated file from the early stages of the GUI and is no longer in use. I will delete it at some point. SliceSelection was an initial attempt at implementing slice selection, but I did not continue with that approach. Apologies for any confusion.sequenceCalibration.py 是 GUI 早期阶段的过时文件，不再使用。我会在某个时候删除它。SliceSelection 是实现切片选择的初步尝试，但我没有继续采用这种方法。对于任何混淆，我们深表歉意。↳
thank u.Will there be a 2d GRE sequence in your plan? I'm an amateur and tried to write it myself, but I failed.

**Reply by volatile-static** [reply]
> > sequenceCalibration.py is an outdated file from the early stages of the GUI and is no longer in use. I will delete it at some point. SliceSelection was an initial attempt at implementing slice selection, but I did not continue with that approach. Apologies for any confusion.
> 
> thank u.Will there be a 2d GRE sequence in your plan? I'm an amateur and tried to write it myself, but I failed.

We'd tried selective RF pulse on a 2D GRE sequence last year, but the SNR is so low that we deprecated it.
FYR: https://github.com/volatile-static/MaRGEx/blob/xt/seq/gre2d5.py

**Reply by josalggui** [reply]
> > > sequenceCalibration.py is an outdated file from the early stages of the GUI and is no longer in use. I will delete it at some point. SliceSelection was an initial attempt at implementing slice selection, but I did not continue with that approach. Apologies for any confusion.
> 
> > 
> 
> > thank u.Will there be a 2d GRE sequence in your plan? I'm an amateur and tried to write it myself, but I failed.
> 
> 
> 
> We'd tried selective RF pulse on a 2D GRE sequence last year, but the SNR is so low that we deprecated it.
> 
> FYR: https://github.com/volatile-static/MaRGEx/blob/xt/seq/gre2d5.py

No, at low field I do not use to work with 2D due to SNR issues.
